### PR TITLE
creating cron-workflow to invoke job template for the sync-job

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
## Related Issues and Dependencies

Partially Addresses: https://github.com/thoth-station/thoth-application/issues/2513
Related to: https://github.com/thoth-station/sync-job/pull/61

## Does this require new deployment ?

- [ ] Deployment for Test and Stage `AICoE/aicoe-cd` and Prod `operate-first/argocd-apps`.

## Description

Creating a cron-workflow for the sync-job

/cc @harshad16

### CODE REVIEW

**Comment by Gregory-Pereira** [comment]
I know that when going from stage to production we should be syncing adviser, revsolver, and security-indicator documents, but what about from production to stage?

File: sync-job/base/cron-worfklow.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  concurrencyPolicy: "Replace"
+  startingDeadlineSeconds: 0
+  workflowSpec:
+    paralellism: 4
+    entrypoint: sync-job-wt
+    templates:
+      container:
+        image: quay.io/thoth-station/sync-job:v0.2.3-dev
+        command: ["oc", "process"]
+        args: ["-p IMAGE_VERSION=$IMAGE_VERSION", "-p THOTH_SYNC_FORCE=$THOTH_SYNC_FORCE", "-p THOTH_SYNC_GRACEFUL=$THOTH_SYNC_GRACEFUL", "-p THOTH_SYNC_DEBUG=$THOTH_SYNC_DEBUG", "-f openshift.yaml", " | oc apply -f -"]
```
Comment (harshad16) [inline sync-job/base/cron-worfklow.yaml:17]: This is not needed.
There is concept of source-to-image , the entrypoint is already set to the run script that can `app.py` and others based on the programming language. The pointer to different script can be done via setting env vars
https://docs.openshift.com/dedicated/3/using_images/s2i_images/python.html#using-images-python-configuration

File: sync-job/base/cron-worfklow.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        env:
+          - name: IMAGE_VERSION
+            value: v0.2.3-dev
```
Comment (harshad16) [inline sync-job/base/cron-worfklow.yaml:20]: This is not needed.
this was used on the sync-job template and is not a part of the application itself. 
https://github.com/thoth-station/sync-job/blob/f630e32b9b62b0e81879a16f95ea34b7e1dadce8/openshift.yaml#L79

File: sync-job/base/openshift.yaml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                    secretKeyRef:
+                      name: postgresql
+                      key: database-name
+              resources:
+                limits:
+                  memory: "2Gi"
+                  cpu: 4
+                requests:
+                  memory: "2Gi"
+                  cpu: 4
```
Comment (harshad16) [inline sync-job/base/openshift.yaml:161]: The workflow would do the work of the job, 
we won't require a separate job.
the cronworkflow would schedule workflow as needed. 

Move all these env vars to the cronworkflow.

**Review by harshad16 — CHANGES_REQUESTED** [review]
good start
i have added some suggestions.

**Comment by Gregory-Pereira** [comment]
It uses 3 templates because both for prod and stage, there are 3 separate types of documents to be synced. Additionally I can apply the resources (it is correct YAML syntax), however, the workflow controller rejects this `CronWorkflow` by giving me the following error: `"Unable to convert unstructured to CronWorkflow when syncing CronWorkflows" error="cannot restore slice from map"`

**Comment by Gregory-Pereira** [comment]
Cron-workflow is now accepted, ready for review.

**Comment by Gregory-Pereira** [comment]
Attempting to implement this with a cron-workflow which uses a workflow template so the changes be used to both invoke a cronworkflow to run sync-job periodically, as well as be invoked from the management api via workflow. That being said I am encountering an Argo issue: `template reference exceeded max depth (10)`.

**Comment by Gregory-Pereira** [comment]
Issue about template reference exceeding max depth has been resolved, hoping that @harshad16 could take a look when you have time.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve
thanks :100:

**Pull request merged**